src/sentry/eventstream/kafka/backend.py

```diff
@@ -334,12 +394,33 @@ def _get_task_kwargs_and_dispatch(self, message) -> None:
             with metrics.timer("eventstream.duration", instance="dispatch_post_process_group_task"):
                 self._dispatch_post_process_group_task(**task_kwargs)
 
-    @contextmanager
-    def sampled_eventstream_timer(self, instance: str) -> Generator[None, None, None]:
+    def run_post_process_forwarder(
+        self,
+        consumer_group,
+        commit_log_topic,
+        synchronize_commit_group,
+        commit_batch_size=100,
+        commit_batch_timeout_ms=5000,
+        initial_offset_reset="latest",
+    ):
+        logger.debug("Starting post-process forwarder...")
 
-        record_metric = random.random() < 0.1
-        if record_metric is True:
-            with metrics.timer("eventstream.duration", instance=instance):
-                yield
+        if settings.SENTRY_POST_PROCESS_FORWARDER_BATCHING:
```

Comment:
Does it make more sense to have this as an option instead of a setting? So we don't need to deploy to enable/disable it?

https://github.com/getsentry/sentry/blob/master/src/sentry/options/defaults.py#L212

Reply:
The current plan is to deploy the change with the default settings on production(which means streaming consumer). In Sentry 4 Sentry, we will change the settings to True. There is no need for having the ability to change setting during runtime. We want ability to pick up some configuration during startup and use it.